clang/test/Driver/sycl-xarch.cpp

```diff
@@ -0,0 +1,120 @@
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_DEVICE_OPTIONS2
+// RUN: %clangxx -fsycl %s -Xarch_device "-fsanitize=address -DXARCH_DEVICE_TEST -mllvm -enable-merge-functions" -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_DEVICE_OPTIONS3
+// SYCL_XARCH_DEVICE_OPTIONS1: clang{{.*}} "-fsycl-is-device"
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: -fsanitize=address
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: -fsanitize-address-use-after-return=never
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: -fno-sanitize-address-use-after-scope
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: "-mllvm" "-asan-instrumentation-with-call-threshold=0"
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: "-mllvm" "-asan-stack=0"
+// SYCL_XARCH_DEVICE_OPTIONS1-SAME: "-mllvm" "-asan-globals=0"
+// SYCL_XARCH_DEVICE_OPTIONS2: clang{{.*}} "-fsycl-is-device"
+// SYCL_XARCH_DEVICE_OPTIONS2-SAME: XARCH_DEVICE_TEST
+// SYCL_XARCH_DEVICE_OPTIONS3: clang{{.*}} "-fsycl-is-device"
+// SYCL_XARCH_DEVICE_OPTIONS3-SAME: "-mllvm" "-enable-merge-functions"
+
+
+/// test behavior of -Xarch_host with 1 option for SYCL compiler, the flag
+/// should be passed to host compilation only.
+// RUN: %clangxx -fsycl %s -Xarch_host -fsanitize=address -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_HOST_OPTION
+// RUN: %clangxx -fsycl %s -Xarch_host -fsanitize=address -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_HOST_ONLY
+// SYCL_XARCH_HOST_OPTION: clang{{.*}} "-fsycl-is-host"
+// SYCL_XARCH_HOST_OPTION-SAME: -fsanitize=address
+// SYCL_XARCH_HOST_OPTION-SAME: -fsanitize-address-use-after-scope
+// SYCL_XARCH_HOST_OPTION-NEXT: libclang_rt.asan
+// SYCL_XARCH_HOST_ONLY: clang{{.*}} "-fsycl-is-device"
+// SYCL_XARCH_HOST_ONLY-NOT: -fsanitize=address
+// SYCL_XARCH_HOST_ONLY: clang{{.*}} "-fsycl-is-host"
+
+/// test behavior of -Xarch_host with multiple options for SYCL compiler, the
+/// flags should be passed to host compilation only.
+// RUN: %clangxx -fsycl %s -Xarch_host "-fsanitize=address -DXARCH_HOST_TEST -mllvm -enable-merge-functions" -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_HOST_OPTIONS1
+// RUN: %clangxx -fsycl %s -Xarch_host "-fsanitize=address -DXARCH_HOST_TEST -mllvm -enable-merge-functions" -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_HOST_OPTIONS2
+// RUN: %clangxx -fsycl %s -Xarch_host "-fsanitize=address -DXARCH_HOST_TEST -mllvm -enable-merge-functions" -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_XARCH_HOST_OPTIONS3
+// SYCL_XARCH_HOST_OPTIONS1: clang{{.*}} "-fsycl-is-host"
+// SYCL_XARCH_HOST_OPTIONS1-SAME: -fsanitize=address
+// SYCL_XARCH_HOST_OPTIONS1-SAME: -fsanitize-address-use-after-scope
+// SYCL_XARCH_HOST_OPTIONS2: clang{{.*}} "-fsycl-is-host"
+// SYCL_XARCH_HOST_OPTIONS2-SAME: XARCH_HOST_TEST
+// SYCL_XARCH_HOST_OPTIONS3: clang{{.*}} "-fsycl-is-host"
+// SYCL_XARCH_HOST_OPTIONS3-SAME: "-mllvm" "-enable-merge-functions"
```

Comment:
Could you add a test that verifies `-Xarch_host "-mllvm -enable-merge-functions" -Xarch_host -DFOO` behaviors (i.e. passing multiple -Xarch_host or -Xarch_device options on the same command line)

Reply:
Hi, @mdtoguchi 
Done. Thanks very much.